--- yash/ya.sh	2020-11-26 16:31:29.000000000 +0100
+++ ya.sh	2020-11-26 17:29:27.000000000 +0100
@@ -46,16 +46,15 @@
 =cut
 
 yashLog() {
-  printf ":: [ %(%T)T ] :: [ %s ] :: " -1 "${2:-" LOG "}" >&2
-  echo -e "$1" >&2
+  rlLog "$1"
 }
 
 yashLogDebug() {
-  [[ -n "$DEBUG" ]] && yashLog "${FUNCNAME[1]}(): $1" "DEBUG"
+  rlLogDebug "${FUNCNAME[1]}(): $1"
 }
 
 yashLogError() {
-  yashLog "${FUNCNAME[1]}(): $1" "ERROR"
+  rlLogError "${FUNCNAME[1]}(): $1"
 }
 
 __INTERNAL_yash_get_next() {
@@ -314,11 +313,11 @@
 : <<'=cut'
 =pod
 
-=head3 yash_parse
+=head3 rlYash_parse
 
 Parse yaml data to the associative array.
 
-    yash_parse VAR_NAME YAML_DATA
+    rlYash_parse VAR_NAME YAML_DATA
 
 =over
 
@@ -336,7 +335,7 @@
 
 =cut
 
-yash_parse() {
+rlYash_parse() {
   local yaml_data item key value data_type item_type item_type_prev prefix="$3" index=0 yaml_name="$1" res=0
   __INTERNAL_yash_clean yaml_data "$2"
 
@@ -365,7 +364,7 @@
     }
     if [[ "$data_type" == "struct" ]]; then
       yashLogDebug "_____________________________"
-      yash_parse "$yaml_name" "$value" "$prefix$key." || return 1
+      rlYash_parse "$yaml_name" "$value" "$prefix$key." || return 1
     fi
   done
 }
